### ISSUE TITLE
Allow restapi object to modify the headers

### ISSUE THREAD

**Issue by vpitest11** [Mastercard/terraform-provider-restapi#138]
I would like to dynamically add headers relative to attributes of other resources defined in the code, like adding a header with an api key that's not known until after apply, etc.

**Comment by DRuggeri** [comment]
Thanks for the idea @vpitest11 - can you say more about what you'd like to do? Are you thinking that per-resource headers should be supported such that:
* A header set in the resource is ADDED to the provider's headers
* A header that exists in both the resource and provider OVERRIDES the provider's headers

... that is, you're only seeking addition/modification of headers rather than deletes/changes/etc to headers, right?

Note: the idea makes sense, but adding this is a technical challenge because the base set of headers is set on the underlying [api_client](https://github.com/Mastercard/terraform-provider-restapi/blob/298e1d56e4a61f2897f81ce499aaee255424efda/restapi/api_client.go#L57) which is an object shared across the whole provider (the provider's principles generally says that these kinds of config attributes ought to stay consistent across all requests). This would require a structural change that needs some noodling.

**Comment by vpitest11** [comment]
Correct. I only need to add/change additional headers. I don’t need to modify any of the default headers.

For example, adding a header for a resource key or secret, or reusing the same provider and adding/modifying additional headers based on resource properties.

**Comment by Keonik1** [comment]
@DRuggeri Hello!
Could you please take a look at PR #360 ? This feature is very useful
